services/worker-manager/src/providers/aws.js

```diff
@@ -82,83 +64,108 @@ class AwsProvider extends Provider {
       });
     }
 
-    // Make sure we don't get "The same resource type may not be specified more than once in tag specifications" errors
-    const TagSpecifications = config.launchConfig.TagSpecifications ? config.launchConfig.TagSpecifications : [];
-    const instanceTags = [];
-    const otherTagSpecs = [];
-    TagSpecifications.forEach(ts =>
-      ts.ResourceType === 'instance' ? instanceTags.concat(ts.Tags) : otherTagSpecs.push(ts)
-    );
+    const uniqueRegions = new Set(workerPool.config.launchConfigs.map(lc => lc.region));
+    let ec2s = {};
+    uniqueRegions.forEach(region => {
+      ec2s[region] = new aws.EC2({credentials: this.providerConfig.credentials, region});
+    });
 
-    let spawned;
-    try {
-      spawned = await ec2.runInstances({
-        ...config.launchConfig,
-
-        UserData: userData.toString('base64'), // The string needs to be base64-encoded. See the docs above
-
-        MaxCount: toSpawn,
-        MinCount: toSpawn,
-        TagSpecifications: [
-          ...otherTagSpecs,
-          {
-            ResourceType: 'instance',
-            Tags: [
-              ...instanceTags,
-              {
-                Key: 'CreatedBy',
-                Value: `taskcluster-wm-${this.providerId}`,
-              }, {
-                Key: 'Owner',
-                Value: workerPool.owner,
-              },
-              {
-                Key: 'ManagedBy',
-                Value: 'taskcluster',
-              },
-              {
-                Key: 'Name',
-                Value: `${workerPoolId}`,
-              }],
-          },
-        ],
-      }).promise();
-    } catch (e) {
-      this.monitor.err(`Error calling AWS API: ${e}`);
+    for await (let config of workerPool.config.launchConfigs) {
```

Comment:
Isn't this pretty much going to pick the first launchConfig every time? We probably need some randomness or something involved here.